service/src/main/java/com/theokanning/openai/service/ChatFunctionCallArgumentsSerializerAndDeserializer.java

```diff
@@ -32,33 +40,40 @@ public void serialize(JsonNode value, JsonGenerator gen, SerializerProvider seri
         }
     }
 
+    public abstract static class JsonNodeHandler {
+        public abstract JsonNode handle(JsonParser p, DeserializationContext ctxt) throws IOException;
+    }
+
+    public static class MissingNodeHandler extends JsonNodeHandler {
+        @Override
+        public JsonNode handle(JsonParser p, DeserializationContext ctxt) {
+            return JsonNodeFactory.instance.missingNode();
+        }
+    }
+
+    public static class DefaultNodeHandler extends JsonNodeHandler {
+        @Override
+        public JsonNode handle(JsonParser p, DeserializationContext ctxt) throws IOException {
+            return MAPPER.readTree(p);
+        }
+    }
+
     public static class Deserializer extends JsonDeserializer<JsonNode> {
+        private static final Map<JsonToken, JsonNodeHandler> HANDLERS = initializeHandlers();
 
-        private Deserializer() {
+        private static Map<JsonToken, JsonNodeHandler> initializeHandlers() {
+            Map<JsonToken, JsonNodeHandler> handlers = new HashMap<>();
+            handlers.put(JsonToken.VALUE_NULL, new MissingNodeHandler());
+            // Add more handlers for different token types if needed
+            return handlers;
```

Comment:
Nit: Returning collections such as Map - from functions, should preserve their immutability. 
Thing about collections.unmodifiableMap
https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#unmodifiableMap-java.util.Map-